packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/ErrorCode.kt

```diff
@@ -0,0 +1,173 @@
+    RLM_ERR_INCOMING_WEBHOOK_ALREADY_EXISTS("IncomingWebhookAlreadyExists", realm_errno_e.RLM_ERR_INCOMING_WEBHOOK_ALREADY_EXISTS),
+    RLM_ERR_INCOMING_WEBHOOK_DUPLICATE_NAME("IncomingWebhookDuplicateName", realm_errno_e.RLM_ERR_INCOMING_WEBHOOK_DUPLICATE_NAME),
+    RLM_ERR_RULE_NOT_FOUND("RuleNotFound", realm_errno_e.RLM_ERR_RULE_NOT_FOUND),
+    RLM_ERR_API_KEY_NOT_FOUND("ApiKeyNotFound", realm_errno_e.RLM_ERR_API_KEY_NOT_FOUND),
+    RLM_ERR_RULE_ALREADY_EXISTS("RuleAlreadyExists", realm_errno_e.RLM_ERR_RULE_ALREADY_EXISTS),
+    RLM_ERR_RULE_DUPLICATE_NAME("RuleDuplicateName", realm_errno_e.RLM_ERR_RULE_DUPLICATE_NAME),
+    RLM_ERR_AUTH_PROVIDER_DUPLICATE_NAME("AuthProviderDuplicateName", realm_errno_e.RLM_ERR_AUTH_PROVIDER_DUPLICATE_NAME),
+    RLM_ERR_RESTRICTED_HOST("RestrictedHost", realm_errno_e.RLM_ERR_RESTRICTED_HOST),
+    RLM_ERR_API_KEY_ALREADY_EXISTS("ApiKeyAlreadyExists", realm_errno_e.RLM_ERR_API_KEY_ALREADY_EXISTS),
+    RLM_ERR_INCOMING_WEBHOOK_AUTH_FAILED("IncomingWebhookAuthFailed", realm_errno_e.RLM_ERR_INCOMING_WEBHOOK_AUTH_FAILED),
+    RLM_ERR_EXECUTION_TIME_LIMIT_EXCEEDED("ExecutionTimeLimitExceeded", realm_errno_e.RLM_ERR_EXECUTION_TIME_LIMIT_EXCEEDED),
+    RLM_ERR_NOT_CALLABLE("NotCallable", realm_errno_e.RLM_ERR_NOT_CALLABLE),
+    RLM_ERR_USER_ALREADY_CONFIRMED("UserAlreadyConfirmed", realm_errno_e.RLM_ERR_USER_ALREADY_CONFIRMED),
+    RLM_ERR_USER_NOT_FOUND("UserNotFound", realm_errno_e.RLM_ERR_USER_NOT_FOUND),
+    RLM_ERR_USER_DISABLED("UserDisabled", realm_errno_e.RLM_ERR_USER_DISABLED),
+    RLM_ERR_AUTH_ERROR("AuthError", realm_errno_e.RLM_ERR_AUTH_ERROR),
+    RLM_ERR_BAD_REQUEST("BadRequest", realm_errno_e.RLM_ERR_BAD_REQUEST),
+    RLM_ERR_ACCOUNT_NAME_IN_USE("AccountNameInUse", realm_errno_e.RLM_ERR_ACCOUNT_NAME_IN_USE),
+    RLM_ERR_INVALID_PASSWORD("InvalidPassword", realm_errno_e.RLM_ERR_INVALID_PASSWORD),
+    RLM_ERR_SCHEMA_VALIDATION_FAILED_WRITE("SchemaValidationFailedWrite", realm_errno_e.RLM_ERR_SCHEMA_VALIDATION_FAILED_WRITE),
+    RLM_ERR_APP_UNKNOWN("Unknown", realm_errno_e.RLM_ERR_APP_UNKNOWN),
+    RLM_ERR_MAINTENANCE_IN_PROGRESS("MaintenanceInProgress", realm_errno_e.RLM_ERR_MAINTENANCE_IN_PROGRESS),
+    RLM_ERR_USERPASS_TOKEN_INVALID("UserpassTokenInvalid", realm_errno_e.RLM_ERR_USERPASS_TOKEN_INVALID),
+    RLM_ERR_WEBSOCKET_RESOLVE_FAILED_ERROR("ResolveFailedError", realm_errno_e.RLM_ERR_WEBSOCKET_RESOLVE_FAILED_ERROR),
+    RLM_ERR_WEBSOCKET_CONNECTION_CLOSED_CLIENT_ERROR("ConnectionClosedClientError", realm_errno_e.RLM_ERR_WEBSOCKET_CONNECTION_CLOSED_CLIENT_ERROR),
+    RLM_ERR_WEBSOCKET_CONNECTION_CLOSED_SERVER_ERROR("ConnectionClosedServerError", realm_errno_e.RLM_ERR_WEBSOCKET_CONNECTION_CLOSED_SERVER_ERROR),
+    RLM_ERR_CALLBACK("Callback", realm_errno_e.RLM_ERR_CALLBACK),
+    RLM_ERR_UNKNOWN("Unknown", realm_errno_e.RLM_ERR_UNKNOWN);
+
+    actual companion object {
+        actual fun of(nativeValue: Int): ErrorCode =
+            values().first() { value ->
```

Comment:
We can add this failsafe mechanism, however, it is not necessary because the Core error codes are a close set of values, and we have tests that validate we map all values.

Reply:
True, I would prefer to be totally safe here though. Since if this ever crashes, it would obfuscate the real error, which is even more annoying.

Reply:
Yup, that would be better 👍